src/python/pants/jvm/target_types.py

```diff
@@ -51,10 +52,18 @@ class JvmArtifactUrlField(StringField):
     help = (
         "A URL that points to the location of this artifact. If specified, Pants will not fetch this artifact "
         "from default maven repositories, and instead fetch the artifact from this URL. To use default maven "
-        "repositories, do not set this value. \n\nNote that `file:` URLs are not presently supported."
+        "repositories, do not set this value. \n\nNote that `file:` URLs are not supported due to Pants' "
+        "sandboxing feature. To use a local `JAR` file, use the `jar` field instead."
     )
 
 
+class JvmArtifactJarSourceField(SingleSourceField):
+    alias = "jar"
+    expected_file_extensions = (".jar",)
+    required = False
+    help = "A JAR file that provides this artifact to the lockfile resolver, instead of a maven repository."
```

Comment:
Should mention that this is a file path relative to the target.

Also, should probably add a `Target.validate` implementation to `JarArtifact`: https://github.com/pantsbuild/pants/blob/c2f7674375561040a47ee5bece55441d90a30711/src/python/pants/engine/target.py#L600-L606

Reply:
Noted.